src/components/auth/RegisterStudent.vue

```diff
@@ -529,31 +528,32 @@ const handleFormSubmit = async (isFormValid) => {
 };
 
 const validateCode = async (studentCode, outerIndex = 0) => {
-  if (studentCode && studentCode !== '') {
-    const activationCode = await fetchDocById('activationCodes', studentCode, undefined, 'admin', true, true).catch(
-      (error) => {
-        errors.value = error;
-        dialogMessage.value =
-          'The code does not belong to any organization \n Please enter a valid code or select: \n "I don`t have code "';
-        showErrorDialog();
-        submitted.value = false;
-        return null;
-      },
-    );
-    if (activationCode.orgId && errors.value === '') {
+  // @TODO: Add proper error handling.
+  if (!studentCode || studentCode === '') return;
+
+  try {
+    const activationCode = await fetchDocById('activationCodes', studentCode, undefined, 'admin', true, true);
+
+    if (activationCode.orgId) {
       state.students[outerIndex].orgName = `${_capitalize(activationCode.orgType)} - ${
         activationCode.orgName ?? activationCode.orgId
       }`;
       state.students[outerIndex].activationCode = studentCode;
       orgName.value = `${_capitalize(activationCode.orgType)} - ${activationCode.orgName ?? activationCode.orgId}`;
+    }
+  } catch (error) {
+    console.error('Failed to validate activation code', error);
+
+    if (!state.students[outerIndex].noActivationCode || props.code) {
+      dialogMessage.value = `The code ${studentCode} does not belong to any organization \n please enter a valid code or select: "I do not have code"`;
     } else {
-      errors.value = '';
-      if (!state.students[outerIndex].noActivationCode || props.code) {
-        dialogMessage.value = `The code ${studentCode} does not belong to any organization \n please enter a valid code or select: "I do not have code"`;
-        showErrorDialog();
-      }
-      return false;
+      dialogMessage.value =
+        'The code does not belong to any organization \n Please enter a valid code or select: \n "I don`t have code "';
```

Comment:
```suggestion
        'The code does not belong to any organization \n Please enter a valid code or select: \n "I do not have a code "';
```
Another small grammer change.

We might also have to change the text on the button or whatever the user is upposed to be selecting in this case?

Reply:
Updated the copy in both cases to match the form checkbox